tfhe/src/core_crypto/fft_impl/fft64/math/fft/x86.rs

```diff
@@ -685,7 +683,10 @@ pub fn convert_add_backward_torus_u64_v4(
             debug_assert_eq!(n, twisties.im.len());
 
             let normalization = avx512f._mm512_set1_pd(1.0 / n as f64);
-            let scaling = avx512f._mm512_set1_pd(2.0_f64.powi(u64::BITS as i32));
+            // cursed: passing this through black_box prevents the compiler from loading the
+            // constant from memory inside the loop
+            let scaling =
+                core::hint::black_box(avx512f._mm512_set1_pd(2.0_f64.powi(u64::BITS as i32)));
```

Comment:
what is the compiler doing otherwise ?

also, worth a report of the issue ?

Reply:
this is an llvm issue so it'll probably take forever to resolve, if at all

Reply:
I understand but, the guys in the rust project have been reactive and even if it does not get solved right away it's good to open the issue, reference it here and have a way to follow up :) they have had some llvm bug fixed from some of our reports !

Reply:
removed the black box as I could not see any difference in the generated assembly and the only data loaded during the loop were inputs + twiddles

https://github.com/zama-ai/tfhe-rs-internal/issues/694